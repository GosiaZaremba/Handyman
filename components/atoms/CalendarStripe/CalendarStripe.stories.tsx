import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {CalendarStripe} from './CalendarStripe';

storiesOf('atoms/CalendarStripe', module)
  .addDecorator(story => story())
  .add('default', () => <CalendarStripe />);
