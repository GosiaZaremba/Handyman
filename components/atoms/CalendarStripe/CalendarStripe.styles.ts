import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  calendar: {
    marginTop: 10,
    height: 50,
  },
  dateNumberStyle: {
    color: Colors.font.darkGrey,
    fontFamily: 'OpenSans-Regular',
    fontSize: 12,
  },
  dateNameStyle: {
    color: Colors.font.darkGrey,
    fontFamily: 'OpenSans-Regular',
    fontSize: 12,
    marginBottom: 5,
  },
  iconContainer: {
    display: 'none',
  },
  highlightDateNumberStyle: {
    fontFamily: 'OpenSans-Regular',
    color: Colors.font.white,
    fontSize: 12,
  },
  highlightDateNameStyle: {
    color: Colors.font.darkGrey,
    fontFamily: 'OpenSans-Regular',
    fontSize: 12,
    marginTop: 5,
  },
  highlightDateNumberContainerStyle: {
    backgroundColor: Colors.calendar,
    borderRadius: 25,
    height: 28,
    width: 28,
    justifyContent: 'center',
    elevation: 8,
  },
  dayContainerStyle: {
    height: 32,
  },
  innerStyle: {flex: 1},
});
