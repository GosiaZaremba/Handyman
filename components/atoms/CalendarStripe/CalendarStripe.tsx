/* eslint-disable @typescript-eslint/no-explicit-any */
import React, {useImperativeHandle, useRef, useState, forwardRef} from 'react';
import {View} from 'react-native';
import ReactNativeCalendarStrip from 'react-native-calendar-strip';
import CalendarStrip from 'react-native-calendar-strip';
import styles from './CalendarStripe.styles';

export type SelectedDateReference = {
  getValue: () => undefined | Date | string;
};

export type Props = {
  testID?: string;
};

const CalendarStripeWithReference: React.ForwardRefRenderFunction<
  SelectedDateReference,
  Props
> = ({testID}, ref) => {
  const [selectedDate, setSelectedDate] = useState<Date>(new Date());

  const dateReference = useRef<ReactNativeCalendarStrip>(null);

  const onDatePick = (date: any) => {
    setSelectedDate(date);
  };

  useImperativeHandle(ref, () => ({
    getValue: () => {
      return selectedDate;
    },
  }));
  return (
    <View testID={testID}>
      <CalendarStrip
        scrollable={true}
        showMonth={false}
        scrollerPaging={true}
        style={styles.calendar}
        innerStyle={styles.innerStyle}
        dateNumberStyle={styles.dateNumberStyle}
        dateNameStyle={styles.dateNameStyle}
        iconContainer={styles.iconContainer}
        startingDate={new Date()}
        selectedDate={selectedDate}
        highlightDateNumberStyle={styles.highlightDateNumberStyle}
        highlightDateNumberContainerStyle={
          styles.highlightDateNumberContainerStyle
        }
        highlightDateNameStyle={styles.highlightDateNameStyle}
        dayContainerStyle={styles.dayContainerStyle}
        ref={dateReference}
        onDateSelected={date => onDatePick(date)}
      />
    </View>
  );
};

export const CalendarStripe = forwardRef(CalendarStripeWithReference);
