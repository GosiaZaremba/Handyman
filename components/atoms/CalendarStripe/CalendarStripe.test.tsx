import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {CalendarStripe, SelectedDateReference} from './CalendarStripe';

describe('Atom calendarStripe', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<CalendarStripe testID="atom-calendarStripe" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure date is extracted and passed correctly', () => {
    const calendarRef = React.createRef<SelectedDateReference>();
    const {getByTestId} = render(
      <CalendarStripe testID="atom-calendarStripe" ref={calendarRef} />,
    );
    const calendarStripe = getByTestId('atom-calendarStripe');
    fireEvent(
      calendarStripe.props.children,
      'onDateSelected',
      '2023-02-22T20:15:24.192Z',
    );
    expect(calendarRef.current?.getValue()).toBe('2023-02-22T20:15:24.192Z');
  });
});
