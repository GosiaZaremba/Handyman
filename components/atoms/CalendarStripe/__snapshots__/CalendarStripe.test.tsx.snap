// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom calendarStripe Component Snapshot 1`] = `
<View
  testID="atom-calendarStripe"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": undefined,
        },
        Object {
          "height": 50,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "display": "none",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../react-native-calendar-strip/src/img/left-arrow-black.png",
              }
            }
            style={
              Array [
                Object {
                  "resizeMode": "contain",
                },
                Object {
                  "height": 0,
                  "width": 0,
                },
                undefined,
                undefined,
                Object {
                  "opacity": 1,
                },
              ]
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        />
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "display": "none",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../react-native-calendar-strip/src/img/right-arrow-black.png",
              }
            }
            style={
              Array [
                Object {
                  "resizeMode": "contain",
                },
                Object {
                  "height": 0,
                  "width": 0,
                },
                undefined,
                undefined,
                Object {
                  "opacity": 1,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
